autoload/asyncomplete.vim

```diff
@@ -235,8 +235,18 @@ function! s:notify_sources_to_refresh(ctx, force) abort
         endif
         if l:refresh
             try
-                call asyncomplete#log('core.s:notify_sources_to_refresh', 'completor()', l:source_name, a:ctx)
-                call s:sources[l:source_name].completor(s:sources[l:source_name], a:ctx)
+                let l:ctx = a:ctx
+                let l:typed = l:ctx['typed']
+                let l:matchpos = s:get_matchpos(s:sources[l:source_name], a:ctx)
```

Comment:
we are recomputing some of these, can you only compute if it hasn't been computed yet? You can most likely use `if !exits('l:matchpos')` to check if has been completed or not.